src/Compiler/Microsoft.AspNetCore.Razor.Language/src/ComparerUtilities.cs

```diff
@@ -9,6 +9,17 @@ namespace Microsoft.AspNetCore.Razor.Language;
 
 internal static class ComparerUtilities
 {
+    public static bool AreDocumentationObjectsEqual(object? x, object? y)
+    {
+        return (x, y) switch
+        {
+            (string s1, string s2) => s1 == s2,
+            (DocumentationDescriptor d1, DocumentationDescriptor d2) => d1.Equals(d2),
+            (null, null) => true,
+            _ => false
```

Comment:
I'd somewhat prefer this case just throws. We don't expose public way to get into a different case, right? Or is the concern that you might have `string` compared to `DocumentationDescriptor`? Maybe a catchall could be `(string or DocumentationDescriptor or null, string or DocumenationDescriptor or null)`?

Reply:
> We don't expose public way to get into a different case, right?

Correct.

> Or is the concern that you might have string compared to DocumentationDescriptor? 

You've got it.

> I'd somewhat prefer this case just throws.
> Maybe a catchall could be (string or DocumentationDescriptor or null, string or DocumenationDescriptor or null)?

I agree that this doesn't feel right, and I'll take a look. Thanks for spotting this!